### PR TITLE
Implement db schema v11

### LINKED ISSUE #942: Implement v11 of the DB schema

### PR DESCRIPTION
**Pull request by cjao**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

Closes #942

### CODE REVIEW

File: covalent/_data_store/models.py [hunk 1]
```diff
@@ -60,15 +60,30 @@ class Lattice(Base):
     # Name of the file containing the function string
     function_string_filename = Column(Text)
 
-    # Name of the file containing the serialized executor
-    executor_filename = Column(Text)
+    # Short name describing the executor ("local", "dask", etc)
+    executor = Column(Text)
```
Comment (FyzHsn) [inline covalent/_data_store/models.py:64]: (non-blocking) Wouldn't `executor_name` be more clear?

File: covalent/_results_manager/result.py [hunk 2]
```diff
@@ -774,6 +783,8 @@ def upsert_electron_data(self, db: DataStore):
                 if not isinstance(node_output, TransportableObject):
                     node_output = TransportableObject(node_output)
 
+                executor = tg.get_node_value(node_id, "metadata")["executor"]
+
```
Comment (FyzHsn) [inline covalent/_results_manager/result.py:787]: (non-blocking) un-necessary empty line

File: tests/covalent_tests/results_manager_tests/results_test.py [hunk 3]
```diff
@@ -58,24 +60,30 @@ def db():
     )
 
 
+le = LocalExecutor(log_stdout="/tmp/stdout.log")
+
```
Comment (FyzHsn) [inline tests/covalent_tests/results_manager_tests/results_test.py:64]: (non-blocking) unnecessary empty lines

**Review by FyzHsn — APPROVED** [review]
I've left some trivial comments just to bring them to your attention. Feel free to resolve them as you see fit. This branch is good ter go as soon as the codecov fix gets merged!

File: covalent/_data_store/models.py (hunk 1 above)
Comment (cjao) [reply covalent/_data_store/models.py:64]: I agree that `executor_name` would be more descriptive. However, `executor` aligns with the corresponding electron metadata field name.

File: covalent/_results_manager/result.py (hunk 2 above)
Comment (cjao) [reply covalent/_results_manager/result.py]: Removed

File: tests/covalent_tests/results_manager_tests/results_test.py (hunk 3 above)
Comment (cjao) [reply tests/covalent_tests/results_manager_tests/results_test.py:64]: This one was actually added by pre-commit.

**Review by FyzHsn — APPROVED** [review]

**Pull request merged**